plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -62,18 +63,21 @@ export const MultistepJsonForm = ({
   const handleNext = () =>
     setActiveStep(Math.min(activeStep + 1, steps.length));
   const handleBack = () => setActiveStep(Math.max(activeStep - 1, 0));
-
+  useEffect(() => {
+    onChange(formDataEvent as IChangeEvent);
+  }, [formDataEvent, onChange]);
```

Comment:
Looks like this is causing a rerender loop. https://github.com/backstage/backstage/blob/d815bcf0c9c368d0d2b0d81f8c002f09ec1d84e6/plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx#L89 isn't using `useCallback`.

Perhaps just call the `onChange` directly rather than doing it through `useEffect`?

Reply:
Yes. You normally never want to depend on callback functions in dependency lists.

Another thing that some people do is

```ts
const MyComponent({ onChange }) {
  const onChangeRef = useRef(onChange);
  useEffect(() => {
    onChangeRef.current = onChange;
  }, [onChange]);
}
```

and then use `onChangeRef.current` instead. But not sure if that's necessary here.

Reply:
I've fixed this up.